feat: revert opcm v1 & add opcmv2 op-deployer support

0xniha opened the pull request:
**Description**

Adds OPCM v2 support to the deployment pipeline with a feature flag to conditionally route deployments through v1 or v2. When OPCM v2 is enabled, v1 public functions are disabled.

**Changes:**

- Added `OPCMV2DevFlag` in `op-deployer/pkg/deployer/devfeatures.go`
- OPCM v1 functions (`deploy`, `upgrade`, `upgradeSuperchainConfig`, `addGameType`, `updatePrestate`, `migrate`) revert with `OPContractsManager_V1Disabled()` when v2 is enabled
- `DeployOPChain.s.sol` checks the feature flag and routes to v1 or v2 accordingly
- OPCM v2 config builder supports three dispute game types: `CANNON`, `PERMISSIONED_CANNON`, `CANNON_KONA`
- Converter functions unify v1/v2 outputs to a common `Output` struct
- Validation logic updated to handle both versions
- Added `OpcmV2Impl` and `OpcmContainerImpl` to implementations tracking
- Commented out `deny_warnings` in `foundry.toml` temporarily

0xiamflux committed:
feat: revert opcm v1 on public interface functions (#699)

* feat: revert opcm v1 on public interface functions

* fix: add dev flag check for OPCM v2

0xiamflux committed:
feat: OPCM v2 support on for op-deployer (#701)

* feat: add feature flag initial

* feat: add feature flags to deploy chains using OPCM v2

* chore: remove unused function, update helper function names

* chore: remove unused DeployOPChainV2

* chore: add todo for obtaining a superchain config per deployment

* fix: remove unnecessary arg and replace function visibility

---------

Co-authored-by: niha <205694301+0xniha@users.noreply.github.com>

maurelian commented on packages/contracts-bedrock/scripts/deploy/DeployOPChain.s.sol:
```diff
@@ -81,53 +143,133 @@ contract DeployOPChain is Script {
+    /// @notice Converts Types.DeployOPChainInput to IOPContractsManagerV2.FullConfig.
+    /// @param _input The input parameters.
+    /// @return config_ The deployed input parameters.
+    function toOPCMV2DeployInput(Types.DeployOPChainInput memory _input)
```
interesting that #18079 didn't require this. It looks like opcmv2.deploy() is not tested yet.

0xniha committed:
fix: add superchainConfig input & fix tests in deployopchain (#705)

* fix: add superchainConfig input & fix tests in deployopchain

* docs: add TODOs

maurelian commented on packages/contracts-bedrock/src/L1/opcm/OPContractsManagerV2.sol:
```diff
@@ -121,6 +121,8 @@ contract OPContractsManagerV2 is ISemver {
         // Dispute game configuration.
         DisputeGameConfig[] disputeGameConfigs;
+        // Feature flags.
+        bool useCustomGasToken;
```
I think this should be in its own PR?

0xniha replied:
agree! I'll remove this.

0xniha replied:
fixed: [`a324054`](https://github.com/ethereum-optimism/optimism/pull/18399/commits/a324054f0a5a55039aa83ff0e66081a5ef580138)

maurelian commented:
/ci authorize e805bdc3209656669b70b960e537a09510ff68e4

0xniha committed:
feat: add opcmv2 flag to deployImplementations and apply opcmv2 deployment test

* feat: add opcmv2 flag to deployImplementations and add apply opcmv2 deployment test

* fix: hardcode flag to avoid import cycles & refactor createOPCMContractV2

* fix: remove todo comment

* fix: deploy implementations test branching

* refactor: add deployOPCMStandardValidatorV2

* fix: add test and script checks

0xiamflux committed:
Merge branch 'develop' into sync/opcm2

Conflicts:
- op-deployer/pkg/deployer/devfeatures.go
- packages/contracts-bedrock/scripts/deploy/DeployImplementations.s.sol
- packages/contracts-bedrock/src/L1/opcm/OPContractsManagerV2.sol

0xiamflux committed:
Merge pull request #732 from defi-wonderland/sync/opcm2

chore: sync opcm2-revert-opcm with develop

0xniha commented on op-deployer/pkg/deployer/pipeline/opchain.go:
```diff
@@ -104,6 +104,15 @@ func makeDCI(intent *state.Intent, thisIntent *state.ChainIntent, chainID common
 	}
 
+	// Select which OPCM to use based on dev feature flag
+	opcmAddr := st.ImplementationsDeployment.OpcmImpl
+	if devFeatureBitmap, ok := intent.GlobalDeployOverrides["devFeatureBitmap"].(common.Hash); ok {
+		opcmV2Flag := common.HexToHash("0x0000000000000000000000000000000000000000000000000000000000010000")
+		if isDevFeatureEnabled(devFeatureBitmap, opcmV2Flag) && st.ImplementationsDeployment.OpcmV2Impl != (common.Address{}) {
+			opcmAddr = st.ImplementationsDeployment.OpcmV2Impl
+		}
+	}
```
Uses `isDevFeatureEnabled`, that mirrors the function in `devfeatures.go`, instead of `deployer.isDevFeatureEnabled` to avoid import cycles.

Is there a way to fix this?

smartcontracts replied:
I guess could live in a dedicated utils file? But honestly I feel like it doesn't matter all that much

0xniha commented on op-deployer/pkg/deployer/devfeatures.go:
```diff
@@ -17,6 +17,12 @@ var (
 	DeployV2DisputeGamesDevFlag = common.HexToHash("0x0000000000000000000000000000000000000000000000000000000000000100")
+
+	// CustomGasTokenDevFlag enables the custom gas token.
+	CustomGasTokenDevFlag = common.HexToHash("0x0000000000000000000000000000000000000000000000000000000000001000")
```
Need to remove this CGT flag

0xniha replied:
fixed: [`a324054`](https://github.com/ethereum-optimism/optimism/pull/18399/commits/a324054f0a5a55039aa83ff0e66081a5ef580138)

0xniha commented on packages/contracts-bedrock/interfaces/L1/opcm/IOPContractsManagerV2.sol:
```diff
@@ -81,6 +81,7 @@ interface IOPContractsManagerV2 {
         uint256 l2ChainId;
         IResourceMetering.ResourceConfig resourceConfig;
         DisputeGameConfig[] disputeGameConfigs;
+        bool useCustomGasToken;
```
Remove

0xniha replied:
fixed: [`a324054`](https://github.com/ethereum-optimism/optimism/pull/18399/commits/a324054f0a5a55039aa83ff0e66081a5ef580138)

0xniha commented on packages/contracts-bedrock/scripts/deploy/Deploy.s.sol:
```diff
@@ -547,7 +553,8 @@ contract Deploy is Deployer {
             resourceConfig: Constants.DEFAULT_RESOURCE_CONFIG(),
-            disputeGameConfigs: disputeGameConfigs
+            disputeGameConfigs: disputeGameConfigs,
+            useCustomGasToken: cfg.useCustomGasToken()
```
remove

0xniha replied:
fixed: [`a324054`](https://github.com/ethereum-optimism/optimism/pull/18399/commits/a324054f0a5a55039aa83ff0e66081a5ef580138)

0xniha commented on packages/contracts-bedrock/scripts/deploy/ReadImplementationAddresses.s.sol:
```diff
@@ -59,24 +62,53 @@ contract ReadImplementationAddresses is Script {
+
+            // OPCMV2 doesn't expose these addresses directly, so we set them to zero
+            // These are internal to the OPCM container and not meant to be accessed externally
+            output_.opcmGameTypeAdder = address(0);
```
NIT: could add TODO comment for removing this when OPCMv1 is deprecated

0xniha commented on packages/contracts-bedrock/scripts/deploy/ReadImplementationAddresses.s.sol:
```diff
@@ -86,6 +118,12 @@ contract ReadImplementationAddresses is Script {
+        // Both v1 and v2 share the same interface for this function.
+        return IOPContractsManager(_opcmAddr).isDevFeatureEnabled(DevFeatures.OPCM_V2);
+    }
+
```
remove this

0xniha replied:
fixed: [`a324054`](https://github.com/ethereum-optimism/optimism/pull/18399/commits/a324054f0a5a55039aa83ff0e66081a5ef580138)

0xniha commented on packages/contracts-bedrock/src/L1/opcm/OPContractsManagerV2.sol:
```diff
@@ -583,7 +585,16 @@ contract OPContractsManagerV2 is ISemver, OPContractsManagerUtilsCaller {
             ),
-            disputeGameConfigs: _upgradeInput.disputeGameConfigs
+            disputeGameConfigs: _upgradeInput.disputeGameConfigs,
+            useCustomGasToken: abi.decode(
+                _loadBytes(
+                    address(_chainContracts.systemConfig),
+                    _chainContracts.systemConfig.isCustomGasToken.selector,
+                    "overrides.cfg.useCustomGasToken",
+                    _upgradeInput.extraInstructions
+                ),
+                (bool)
+            )
```
remove

0xniha replied:
fixed: [`a324054`](https://github.com/ethereum-optimism/optimism/pull/18399/commits/a324054f0a5a55039aa83ff0e66081a5ef580138)

0xniha commented on packages/contracts-bedrock/src/L1/opcm/OPContractsManagerV2.sol:
```diff
@@ -787,6 +798,11 @@ contract OPContractsManagerV2 is ISemver, OPContractsManagerUtilsCaller {
             );
         }
 
+        // If the custom gas token feature was requested, enable it in the SystemConfig.
+        if (_cfg.useCustomGasToken) {
+            _cts.systemConfig.setFeature(Features.CUSTOM_GAS_TOKEN, true);
+        }
+
```
remove

0xniha replied:
fixed: [`a324054`](https://github.com/ethereum-optimism/optimism/pull/18399/commits/a324054f0a5a55039aa83ff0e66081a5ef580138)

0xniha committed:
fix: remove cgt & fix tests (#736)

* fix: remove cgt

* fix: query chainIdToBatchInboxAddress in chain assertions script

* fix: add skip deploy ocpmv1 when v2 enabled

* refactor: remove unnecesary function

* fix: opcm semver

* chore: add todo comment with issue number (#737)

---------

Co-authored-by: IamFlux <175354924+0xiamflux@users.noreply.github.com>

smartcontracts commented on packages/contracts-bedrock/foundry.toml:
```diff
@@ -72,7 +72,7 @@ fs_permissions = [
 # 5159 error code is selfdestruct error code
 ignored_error_codes = ["transient-storage", "code-size", "init-code-size", "too-many-warnings", 5159]
-deny_warnings = true
+# deny_warnings = true
```
Will need to undo this diff

0xOneTony commented:
/ci authorize https://github.com/ethereum-optimism/optimism/pull/18399/commits/a324054f0a5a55039aa83ff0e66081a5ef580138

0xiamflux committed:
Merge develop into sc-feat/opcm2-revert-opcm

0xiamflux committed:
Merge pull request #744 from defi-wonderland/sync/opcm2

chore: sync develop into sc-feat/opcm2-revert-opcm

0xiamflux commented:
/ci authorize https://github.com/ethereum-optimism/optimism/pull/18399/commits/709037ec50a80375698600a1d0fc3d4b66c8fe62

0xOneTony commented:
/ci authorize https://github.com/ethereum-optimism/optimism/pull/18399/commits/709037ec50a80375698600a1d0fc3d4b66c8fe62

smartcontracts commented on packages/contracts-bedrock/src/L1/OPContractsManager.sol:
```diff
@@ -2306,8 +2306,19 @@ contract OPContractsManager is ISemver {
     // -------- Methods --------
 
+    /// @notice Modifier that reverts if OPCM v2 is enabled via dev feature flag.
+    modifier revertIfV2Enabled() {
```
Technically the "right" thing to do here would be to use a function like `_assertV2NotEnabled` based on the patterns we have in this codebase but because we're killing off this contract I really don't care all that much. Up to you if you want to swap it.

0xiamflux replied:
Can you point me to one of these functions to look at the pattern? I think the best thing would be to make it congruent with the rest of the code while we don't remove the contract.

0xiamflux replied:
Maybe the `_assertOnlyProxyAdminOrProxyAdminOwner` in `ProxyAdminOwnedBase` is a good example?

smartcontracts reviewed (commented):
PR generally seems good. It's a little janky, but that's the nature of the work being done here. We could theoretically spend more time cleaning the PR up but we're going to be forced to do this cleanup anyway when we remove the V1 codebase in the future.

My only ask, let's make sure that the PR that removes the V1 and the V1 feature flag includes some time for dedicated cleanup work so we can leave these functions in a nice state.

0xiamflux committed:
refactor: match assert pattern on OPCM v1 (#749)

* refactor: adheres OPCM v1 revert function to assert pattern

* chore: enable deny_warnings on foundry.toml

* chore: pre-pr ready

smartcontracts commented:
/ci authorize 11d0efa2ca45cbf8317aa3eb2d09d94a577db46c

0xOneTony committed:
Merge branch 'develop' of github.com:defi-wonderland/optimism into sync/opcm2-revert-2

0xiamflux committed:
fix: add implementations for StandardValidator when using OPCMv2

0xOneTony committed:
Merge pull request #757 from defi-wonderland/sync/opcm2-revert-2

chore: opcm2-revert-opcm branch with develop

0xOneTony commented:
/ci authorize https://github.com/ethereum-optimism/optimism/pull/18399/commits/ee078758c222922093ec3acc47f1c0c9aadd3b75

0xiamflux committed:
chore: pre-pr ready (#758)

0xOneTony commented:
/ci authorize https://github.com/ethereum-optimism/optimism/pull/18399/commits/416feae4f6ae078a8d7bb554ea82abef32d14c15

maurelian commented on packages/contracts-bedrock/test/opcm/DeployOPChain.t.sol:
```diff
@@ -214,6 +260,8 @@ contract DeployOPChain_Test is DeployOPChain_TestBase {
 
     function test_customGasToken_enabled_succeeds() public {
+        // TODO(#18536): Remove this when cgt support is added to OPCM v2
+        skipIfDevFeatureEnabled(DevFeatures.OPCM_V2);
```
I think this can be removed now right?

0xOneTony replied:
good catch! removed here https://github.com/ethereum-optimism/optimism/pull/18399/commits/6c9fe7ea8ed8c786fcbf56b9bc84617e11096d60

0xniha committed:
fix: remove cgt checks for opcmv2 (#765)

0xOneTony commented:
/ci authorize https://github.com/ethereum-optimism/optimism/pull/18399/commits/6c9fe7ea8ed8c786fcbf56b9bc84617e11096d60

smartcontracts commented:
/ci authorize 6c9fe7ea8ed8c786fcbf56b9bc84617e11096d60

0xniha committed:
chore: opcm revert develop sync (#769)

* feat(ci): mise; cache & retries. (#18572)

* op-node: Light CL: Always Follow Source using CL (#18571)

* wiring for l2.follow.source

* Follow Safe without handling external safe > local unsafe

* safe follow

* safe follow consider unsafe gap EL Sync

* follow finalized

* Comments

* cleanup

* drop unused interface methods

* sanity check external finalized

* Adjust follow source log level

* op-devstack: Follow Source Support

* op-acceptance-tests: Follow Source

* op-devstack: Follow Source Support

* simplify labeling

* l1 reorg protection: reset

* add reorg tc

* typo

* follow source: check unsafe

* convention

* Add unsafe only reorg test

* devstack/acceptance test : rename FollowSource to FollowL2

* follow upstream source enabled when derivation disabled, reorg detection

* fix unsafe only reorg sync test comments

* rm unused interface method

* dsl

* devstack support for ext sync test + follow l2

* op-acceptance-tests: Follow L2 using Ext + SyncTester

* use blockref

* fix log msg err

* Fix composite interface naming

* op-node: Follow Source: EL / CL support

* devstack: Allow EL or CL to be follow source

* Inject CurrentL1

* logs for injecting CurrentL1

* refactor to less rpc calls

* acceptance-tests: Check CurrentL1 is advancing

* linter

* fix error args

* sanity check external values using L1

* op-node: Follow Source CL and drop unsafeOnly

* op-acceptance-tests: drop EL follow source / Unsafe Only tests

* op-devstack|acceptance-tests: Follow Source CL

* op-node: Follow Client: only allow CL source

* drop unused args

* rm redundant unsafe head reorg logic

* rm unused dsl

* Do not use functions

* chore: Add SafeRename ioutil utility (#18610)

* chore: Add SafeRename ioutil utility

* fix: Lint

* proofs: Update pinned kona client version (#18611)

* feat: have OPCMv2 check upgrade ordering version (#18583)

* feat: have OPCMv2 check upgrade ordering version

This PR introduces a check in OPCM that validates that the
version difference in the previous OPCM and the current OPCM is
acceptable. This PR explicitly disallows skipping OPCM versions
which was previously an implicit requirement.

* fix: properly set version testing flag

* fix: semver lock

* fix: move valid instructions allowance

* fix: semver lock

* fix: better mechanism for checking permitted upgrades

* fix: properly run permitted upgrade sequence tests

* chore: bump opcmv2 to v7 (#18631)

* chore: bump opcmv2 to v7

* fix: correct version in upgrade sequence

* fix: broken tests

* op-chain-ops: Add command to calculate an output root from an EL endpoint. (#18626)

* verify opcm v2 (#18590)

* verify opcm v2

* fixes

* fixes

* some fixes

* some fixes

* some fixes

* fix verifyOPCM v1 tests

* fix verifyOPCM v1 tests

* fix ci checks

* better approach to tests, less diff

* improve test coverage

* fix typo

* fix semver

* Update packages/contracts-bedrock/scripts/deploy/VerifyOPCM.s.sol

Co-authored-by: 0xOneTony <112496816+0xOneTony@users.noreply.github.com>

---------

Co-authored-by: 0xOneTony <112496816+0xOneTony@users.noreply.github.com>

* codeowners(op-deployer): add platforms team as codeowners (#18635)

* ci: Fix contracts test-upgrade test junit output (#18637)

* ci: Fix contracts test-upgrade test junit output

* fix test-upgrade xml output

* proofs: Add provable output roots to super DGs (#18605)

* proofs: Add provable output roots to super DGs

This patch updates the SuperFaultDisputeGame to explicitly prove output
roots contained in the super root. This is accomplished by requiring
proposers to provide the preimage of the super root when creating a
SuperFaultDisputeGame.

The Super DG is then extended with a rootClaim(uint256 _l2ChainId)
method that returns specific output roots contained in the proposal.
This allows withdrawals to be proven using only an output root.

* review comments

* fix nits; update interfaces

* update snapshots

* fix panic in non-l2oo proposals

* op-program: Reduce disk usage of preimage prestate builds in ci (#18641)

* all: upgrade op-geth (#18646)

Pull in superchain registry changes and a few other minor changes.

No upstream merges.

* refactor+fix(op-node/sequencing): l1 origin selection improvements (#18589)

* WIP

* wip

* WIP

* Treat NotFound next L1 origin as chain end

* Use recover mode in sequence window expiry test

* Invoke fault proof earlier and fix typos

Run env.RunFaultProofProgram after computing l2SafeHead
(l2SafeHead.Number.Uint64()/2) and replace the FromGenesis call with
RunFaultProofProgram. Fix minor comment typos and wrap a long log line.

* reduce diff

* Use requireL1OriginAt helper in test

Replace manual error assertions around FindL1Origin with
requireL1OriginAt and remove the now-unused derive import.

* Introduce L2Sequencer.ActMaybeL2StartBlock

* add TestRecoverModeWhenChainHealthy acceptance test

sysgo only

* Add SetSequencerRecoverMode and enable debug logs

* Adjust L1 block time and sequence window test

Set default L1 block time to 12s to match action helper assumptions.
Increase sequencer window size in the test to 50. Compute drift from
UnsafeL2 headers (use UnsafeL2.L1Origin). Simplify L1 mining to always
BatchMineAndSync and remove the extra numL1Blocks > 10 lag guard.

* restore stub

* WIP

* name errs

* refactor

* fix

* add test

* Rename error constant and add L1 origin tests

Rename ErrInvalidL1OriginChild to ErrNextL1OriginOrphaned and adjust the
error text. Add test cases covering L1 reorg and invalid L1 origin, and
refactor the test case construction in the unit test.

* Use drift check for next L1 origin and update tests

Compute driftNext and use it to decide adoption of the next L1 origin
instead of comparing absolute timestamps. Bump MaxSequencerDrift to 1800
and add tests covering max-drift and negative-drift scenarios.

* Refactor L1 origin selection and error handling

Delegate origin decision logic to FindL1OriginOfNextL2Block and handle
synchronous fetch when in recover mode. Remove recover-mode fetching
from CurrentAndNextOrigin and return cached values instead. Update
sequencer error handling to distinguish invalid/orphaned origin (which
triggers a reset) from temporary lookup errors.

* fixes

* fixes

* lint

* don't use pointers

saves some translation code

* handle retries without a "temporary error"

fixes action tests

* use Fjord drift constant

* fix origin_selector_test

mainly just asserting on sentinel errors, and taking account of small
optimization (fewer network calls)

* Simplify FindL1Origin

* move new pure function into a method on los

* Update comment to refer to empty nextL1Origin

* Use errors.Is for L1 origin error checks

* Return L1 origin on validation errors

* Add expectedResult to origin selector tests

Set expectedResult for the test cases with l2Head d1000 and e1000 to
assert the expected L1 origins (c101 and c100 respectively)

* Add assertion message and clarify origin comments

Provide a helpful failure message in sequence_window_expiry_test when
the safe head doesn't advance after the sequencing window expires.
Document that the current L1 origin should always be non-empty and add a
panic guard. Clarify the rationale for requiring the next L1 origin,
include a source link, and note the effect on unsafe block production.

* Store recoverMode and add comment period

* Update op-node/rollup/sequencing/origin_selector.go

Co-authored-by: almanax-ai[bot] <174396398+almanax-ai[bot]@users.noreply.github.com>

* Update op-node/rollup/sequencing/origin_selector.go

Co-authored-by: almanax-ai[bot] <174396398+almanax-ai[bot]@users.noreply.github.com>

---------

Co-authored-by: almanax-ai[bot] <174396398+almanax-ai[bot]@users.noreply.github.com>

* feat: add ReadSuperchainDeployment support for opcm v2 (#18520)

* feat: add readsuperchaindeployment for opcm v2

* fix: argument in PopulateSuperchainState & add docs  (#734)

* fix: set superchainConfig to zero for populateSuperchainState with opcmV1

* docs: add clarification of opcmv1 deprecation

* fix: remove unused opcmV2Enabled var

* test: opcm2 add read sup chain tests (#754)

* chore: expand comment on ReadSuperchainDeployment.s.sol and init.go

* test: add additional test cases for PopulateSuperchainState and InitLiveStrategy

* fix: adds the right number of hex digits to OPCMV2DevFlag in devfeatures.go (#755)

* fix: remove branching in init.go & link TODO issue (#760)

* docs: link todo issue

* refactor: remove branching in init.go WIP

* fix: remove unused opcmv2 feature flag

* fix: remove console logs

* fix: set original timeout, fix isOpcmV2 version and add populateSuperchainState tests

* fix: set original timeout

* fix: isOpcmV2 version

* feat: add test cases for populateSuperchainState and unify tests

* fix: ocpm version comments

---------

Co-authored-by: Flux <175354924+0xiamflux@users.noreply.github.com>
Co-authored-by: OneTony <onetony@defi.sucks>
Co-authored-by: 0xOneTony <112496816+0xOneTony@users.noreply.github.com>

* fix: opcmv2 view function calls

---------

Co-authored-by: Stefano Charissis <stefano@oplabs.co>
Co-authored-by: Changwan Park <pcw109550@gmail.com>
Co-authored-by: Ján Jakub Naništa <jan.jakub.nanista@gmail.com>
Co-authored-by: Inphi <mlaw2501@gmail.com>
Co-authored-by: smartcontracts <14298799+smartcontracts@users.noreply.github.com>
Co-authored-by: Adrian Sutton <adrian@oplabs.co>
Co-authored-by: Michael Amadi <amadimichaeld@gmail.com>
Co-authored-by: 0xOneTony <112496816+0xOneTony@users.noreply.github.com>
Co-authored-by: serpixel <5087962+serpixel@users.noreply.github.com>
Co-authored-by: Josh Klopfenstein <joshklop@oplabs.co>
Co-authored-by: George Knee <georgeknee@googlemail.com>
Co-authored-by: almanax-ai[bot] <174396398+almanax-ai[bot]@users.noreply.github.com>
Co-authored-by: Flux <175354924+0xiamflux@users.noreply.github.com>
Co-authored-by: OneTony <onetony@defi.sucks>

0xOneTony committed:
Revert "chore: opcm revert develop sync (#769)"

This reverts commit d84f0b9dc1fb6592f0a9462f8440acfbb406c1b0.

0xOneTony committed:
Merge branch 'develop' of github.com:defi-wonderland/optimism into sync/opcm2-op-deployer-support

0xOneTony committed:
fix: sync commit

0xOneTony committed:
Merge pull request #774 from defi-wonderland/sync/opcm2-op-deployer-support

fix: sync commit

0xOneTony committed:
Merge branch 'develop' of github.com:defi-wonderland/optimism into sync/opcm2-support-op-deployer

0xOneTony committed:
Merge pull request #775 from defi-wonderland/sync/opcm2-support-op-deployer

chore: sync opcm2 support op deployer with develop
There was a conflict on /DeployOPChain.t.sol

0xOneTony commented:
/ci authorize 871b5d62fb4e10024a4b9d7b41d4eb4fcd1c8801
